File: src/ys_hash.rb

def ys_hash arr
	arr.reduce(8388617) { |sum, n|
		((sum << 1 | sum >> 30) & 0x7fffffff) ^ n
	}.to_s(36)
end

# Test
arr = (0..5000).map { rand 2 ** 8 }
puts ys_hash(arr)

File: src/ys_hash.php
<?

function ys_hash($arr) {
	$h = array_reduce(
		$arr,
		function($sum, $n) {
			return ( ($sum << 1 | $sum >> 30) & 0x7fffffff ) ^ $n;
		},
		8388617
	);

	return base_convert($h, 10, 36);
}

// Test
$arr = array();
for ($i = 0; $i < 5000; $i++) {
	$arr[$i] = rand(0, 255);
}

echo ys_hash($arr) . "\n";

?>